qep-406-markers-hashes-lines-extra-items.md

```diff
@@ -0,0 +1,97 @@
+# QGIS Enhancement: Markers/Hashes line symbol layer extra items
+
+**Date** 2025/12/01
+
+**Author** Julien Cabieces (@troopa81)
+
+**Contact** julien at cabieces at oslandia dot com
+
+**Version** QGIS 4.0
+
+# Summary
+
+This is a follow up of [QEP #345](./qep-345-line-symbollayer-blankareas.md) which provides a tool to define per feature blank segments on marker/hashed line symbol layer where no markers/hashes are displayed (see [here](https://github.com/qgis/QGIS/pull/64047#issue-3659188974) for demo).
+
+The rest of the proposal will mention markers but everything described here applies also to hashes.
+
+This QEP aims at providing tools to create, move, rotate and delete new markers for marker line symbol layer. Those new markers will share the same style as the original markers, the same selected or visibly state. 
+
+## Proposed Solution
+
+The new markers information will be stored per feature, per layer, per symbol layer.
+
+A new Property would be added to the `QgsSymbolLayer` enum and it will be possible to explicity set a data defined property for all `QgsTemplatedLineSymbolLayerBase` symbol layers (which marker line and hash line symbol layer inherits from). It would be possible then to define those new markers either:
+
+- with a specific data field
+- an auxiliary data storage field
+
+This property will be a string of comma separated tuple `(x, y, angle)` where `x` and `y` stands for the marker position in map unit (in the layer CRS)  and `angle` the marker rotation angle in radians.
+
+### UI
+
+A group box `Manual Customization` will integrate both this feature and the blank segments feature.
+
+![](images/qep406/maptool.png)
+
+- User could select a map tool to add item **(1)**: `Left click` would generate a new marker at the given position.
+- User could select a map tool to select already created item to move/delete/rotate them **(2)**: This map tool would follow the principles described [here](https://github.com/qgis/QGIS-Enhancement-Proposals/blob/bbfc1046ff63394e423f2b3b960a99d6734b08b8/qep-344-copy-move-rotate-annotation-layers.md#2-selectmove-annotations) (except for the resize part). Pressing the `Del` button would delete selected markers.
```

Comment:
Would it make sense to highlight or mark the markers to know, which ones are the "normal" ones and which ones are the annotations?

Reply:
Similar to the red lines here https://github.com/qgis/QGIS/pull/64047

Reply:
Done 4d6658c